packages/usa-input-mask/src/index.js

```diff
@@ -1,124 +1,1095 @@
+    for (let i = endPos; i < el.value.length; i += 1) {
+      if (FORMAT_CHARACTERS.indexOf(el.value.charAt(i)) === -1) {
+        endPos = i;
+        break;
+      }
+    }
+
+    el.value =
+      el.value.slice(0, curPos) + el.value.slice(endPos, el.value.length);
+
+    changeValue(inputEl);
+    setCursorPosition(inputEl, curPos);
+  }
+};
+
+/**
+ * Creates and inserts a new shell element around the given input element, and sets various attributes and properties for the input element and shell element.
+ * The shell element consists of a content element with a placeholder text taken from the input element's `PLACEHOLDER` attribute.
+ * The `setInitMask` and `handleValueChange` functions are also called with the input element as an argument.
+ *
+ * @param {HTMLElement} inputEl - The input element to be wrapped in a shell element.
+ */
+const createMaskedInputShell = (inputEl) => {
+  const placeholder = inputEl.getAttribute(`${PLACEHOLDER}`);
+  if (!placeholder) return;
+
+  inputEl.setAttribute("maxlength", placeholder.length);
+  inputEl.setAttribute("data-placeholder", placeholder);
+  inputEl.removeAttribute(`${PLACEHOLDER}`);
 
   const shell = document.createElement("span");
-  shell.classList.add(MASK);
+  shell.classList.add(MASKED_INPUT_SHELL_CLASS);
   shell.setAttribute("data-mask", placeholder);
 
   const content = document.createElement("span");
-  content.classList.add(MASK_CONTENT);
+  content.classList.add(MASKED_INPUT_CONTENT_CLASS);
   content.setAttribute("aria-hidden", "true");
-  content.id = `${input.id}Mask`;
+  content.id = `${inputEl.id}Mask`;
   content.textContent = placeholder;
 
   shell.appendChild(content);
-  input.closest(CONTEXT).insertBefore(shell, input);
-  shell.appendChild(input);
+  inputEl.parentNode.insertBefore(shell, inputEl);
+  shell.appendChild(inputEl);
 };
 
-const setValueOfMask = (el) => {
-  const { value } = el;
-  const placeholderVal = `${el.dataset.placeholder.substr(value.length)}`;
+/**
+ * Returns an object containing the current status message and whether or not the
+ * character type is valid.
+ * @param {HTMLElement} inputEl - The input element.
+ * @param {number} keyCode - The key code of the key that was pressed.
+ * @param {string} key - The character that was entered.
+ * @param {number} curPos - The current position in the input field.
+ * @returns {{currentStatusMessage: string, invalidCharType: boolean}} An object containing the current status message and whether or not the character type is valid.
+ */
+const getMaskMessage = (inputEl, keyCode, key, curPos) => {
+  const isNumber = /^\d+$/.test(key);
+  const invalidAlpha = inputEl.getAttribute("data-invalid-alpha-text")
+    ? inputEl.getAttribute("data-invalid-alpha-text")
+    : "Please enter a letter character here";
+  const invalidNumeric = inputEl.getAttribute("data-invalid-numeric-text")
+    ? inputEl.getAttribute("data-invalid-numeric-text")
+    : "Please enter a number character here";
```

Comment:
- Should the error message specify the user entered an invalid input? Maybe something like
 
> Invalid character. Please enter a _alpha/number_ character here

Reply:
I like this suggestion. I'll incorporate tomorrow unless I see any objections.